client/packages/programs/src/JsonForms/components/Search/SearchWithUserSource.tsx

```diff
@@ -20,33 +21,36 @@ import { useSearchQueries, QueryValues } from './useSearchQueries';
 
 const MIN_CHARS = 3;
 
-const Options = z.object({
-  /**
-   * Which pre-defined query to use (in useSearchQueries)
-   */
-  query: z.enum(QueryValues),
-  /**
-   * Pattern for formatting options list items (e.g. "${firstName} ${lastName}")
-   */
-  optionString: z.string().optional(),
-  /**
-   * Pattern for formatting selected result (as above)
-   */
-  displayString: z.string().optional(),
-  /**
-   * List of fields to save in document data (from selected item object)
-   */
-  saveFields: z.array(z.string()).optional(),
-  /**
-   * Text to show in input field before user entry
-   */
-  placeholderText: z.string().optional(),
-});
+const Options = z
+  .object({
+    source: z.enum(['user']),
```

Comment:
Think `name` or `patient` would be better?

Reply:
"user" means the search parameter comes from user input, as opposed to a document source. It's not talking about _what_ is being searched, but _where_ the "variable" for the search comes from. Hence 'user' and 'document' - but we could use that input to search a document, a patient, or anything.

Reply:
> Nit: the 'document' type is not really a search component, is it? but mare a "clone" component?

Yeah, I'd call it a "lookup" component, strictly speaking. :)

Reply:
> Ah ok I see, you mentioned it. Could you please use a `z.discriminatedUnion` in the parent component though and call 'source' `type`? (there is an existing usage of it which uses type as well). I think it becomes clearer in the code then... You also only have to call zod once in the main component then.

@clemens-msupply I just tried making a zod discriminated union at the top level. It works fine at that level and simplifies the code in the children. But the problem is I still need a named type of each union element in each of the children, otherwise the child components don't know *which* of the union contributor types to expect, and that gets really messy. Also, passing the pre-validated "options" to each child is harder and (even assuming I have two different child types) you have to put an `as` as you pass each "options" down, and I find this really messy and unsatisfying.

That's why I thought just defining a very loose type at the top level and leaving each child to do strict validation on their specifics was simpler.

I can show you what I mean if you still want to pursue it.

Reply:
@clemens-msupply actually I can do it better if I make specific types for each child type and then use them in the discriminated union.

I think this does the trick: c707afb5d

Reply:
cool awesome yes that looks good!!